crates/hyperswitch_connectors/src/connectors/nexixpay.rs

```diff
@@ -311,13 +556,42 @@ impl ConnectorIntegration<PSync, PaymentsSyncData, PaymentsResponseData> for Nex
     }
 }
 
+fn get_payment_id(
+    (metadata, payment_intent): (Option<Value>, Option<nexixpay::NexixpayPaymentIntent>),
+) -> CustomResult<String, errors::ConnectorError> {
+    let connector_metadata = metadata.ok_or(errors::ConnectorError::MissingRequiredField {
+        field_name: "connector_meta",
+    })?;
+    let nexixpay_meta_data =
+        serde_json::from_value::<nexixpay::NexixpayConnectorMetaData>(connector_metadata)
+            .change_context(errors::ConnectorError::ParsingFailed)?;
+    let payment_flow = payment_intent.unwrap_or(nexixpay_meta_data.psync_flow);
+    let payment_id = match payment_flow {
+        nexixpay::NexixpayPaymentIntent::Cancel => nexixpay_meta_data.cancel_operation_id,
+        nexixpay::NexixpayPaymentIntent::Capture => nexixpay_meta_data.capture_operation_id,
+        nexixpay::NexixpayPaymentIntent::Authorize => nexixpay_meta_data.authorization_operation_id,
+    };
+    payment_id.ok_or_else(|| {
+        errors::ConnectorError::MissingRequiredField {
+            field_name: "operation_id",
+        }
+        .into()
+    })
+}
+
 impl ConnectorIntegration<Capture, PaymentsCaptureData, PaymentsResponseData> for Nexixpay {
     fn get_headers(
         &self,
         req: &PaymentsCaptureRouterData,
-        connectors: &Connectors,
+        _connectors: &Connectors,
     ) -> CustomResult<Vec<(String, masking::Maskable<String>)>, errors::ConnectorError> {
-        self.build_headers(req, connectors)
+        let mut header = vec![(
+            headers::IDEMPOTENCY_KEY.to_string(),
+            Uuid::new_v4().to_string().into_masked(),
```

Comment:
Similar thing here (and everywhere else in this file where idempotency key applies), I'd expect the idempotency key to be something to help prevent multiple payments from being successful for the same order if we end up sending multiple requests for the same order. I'd think it should be our payment attempt ID or intent ID, we can consult the documentation.

Reply:
checked with the usage of idempotency in nexixpay. This key is used to club multiple captures or refunds together. 
every capture (in case of multiple manual capture) should have same idempotency key same is the case of refunds. But we cannot pass same idempotency key in refunds and capture. Would using the payment attempt ID or intent ID suffice?

Reply:
as discussed, Idempotency key issue will require core changes where we will be introducing a field to store something like connect-payment-session-id which can we used as idempotency key. We will make this change once we get confirmation change from nexixpay also will add the issue for the same once verified.